x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/type/DataTypes.java

```diff
@@ -177,6 +177,10 @@ public static boolean isDateTime(DataType type) {
         return type == DATETIME;
     }
 
+    public static boolean isNullOrDateTime(DataType type) {
```

Comment:
Since there's already EsqlDataTypes I would remove this method.
Zooming out a bit having both isType and isNullOrType is confusing ? if the null check is so common, why not do it in a generic way instead of having different method variants.

Reply:
👍 
There's also `isNullOrNumeric()`, so i kept the style, but in fact this method's no longer needed, so I removed it.